src/main/java/de/azapps/kafkabackup/common/record/RecordJSONSerde.java

```diff
@@ -43,38 +51,82 @@ public void write(OutputStream outputStream, Record record) throws IOException {
         mapper.writeValue(outputStream, record);
     }
 
-    public static class Deserializer extends StdDeserializer<Record> {
-        public Deserializer() {
+    public static class RecordDeserializer extends StdDeserializer<Record> {
+        public RecordDeserializer() {
             super(Record.class);
         }
 
         @Override
-        public Record deserialize(JsonParser jp, DeserializationContext ctx) throws IOException, JsonProcessingException {
+        public Record deserialize(JsonParser jp, DeserializationContext ctx) throws IOException {
             JsonNode node = jp.getCodec().readTree(jp);
             String topic = node.get(TOPIC_PROPERTY).asText();
             int partition = node.get(PARTITION_PROPERTY).asInt();
             long offset = node.get(OFFSET_PROPERTY).asLong();
+
             String timestampTypeNullableString = node.get(TIMESTAMP_TYPE_PROPERTY).asText(null); // Default seems to be the string "null", which is not wanted here
             TimestampType timestampType = Optional.ofNullable(timestampTypeNullableString)
                     .map(TimestampType::forName)
                     .orElse(TimestampType.NO_TIMESTAMP_TYPE);
-
             // `Long` used instead of primitive `long` to use it as a kind of "optional" with null as a legal value
             Long timestamp = (node.hasNonNull(TIMESTAMP_PROPERTY)) ? (Long) node.get(TIMESTAMP_PROPERTY).asLong() : null;
 
-            String keyBase64 = node.get(KEY_PROPERTY).asText(null); // Default seems to be the string "null", which is not wanted here
-            String valueBase64 = node.get(VALUE_PROPERTY).asText(null);
-            // TODO: parse headers as well
             // TODO: is getting a decoder expensive?
+            String keyBase64 = node.get(KEY_PROPERTY).asText(null); // Default seems to be the string "null", which is not wanted here
             byte[] key = (keyBase64 == null) ? null : Base64.getDecoder().decode(keyBase64);
+            String valueBase64 = node.get(VALUE_PROPERTY).asText(null);
             byte[] value = (valueBase64 == null) ? null : Base64.getDecoder().decode(valueBase64);
 
-            return new Record(topic, partition, key, value, offset, timestamp, timestampType);
+            // Parse headers using registered deserializer
+            JsonNode headersNode = node.findValue(HEADERS_PROPERTY);
+            ConnectHeaders headers;
+            if (headersNode != null) {
+                headers = node.findValue(HEADERS_PROPERTY).traverse(jp.getCodec()).readValueAs(ConnectHeaders.class);
+            } else {
+                headers = new ConnectHeaders();
+            }
```

Comment:
Could use Optional.ofNullable here as well ofcourse!

Reply:
I can't get this to work since readValueAs throws IOException thow... 🤔
```
ConnectHeaders headers = Optional.ofNullable(node.findValue(HEADERS_PROPERTY))
                    .map(n -> n.traverse(jp.getCodec()).readValueAs(ConnectHeaders.class))
                    .orElseGet(() -> new ConnectHeaders());
```

Reply:
Ah throwing RuntimeException totally worked 👍